tests/python/relax/frontend/test_onnx_frontend.py

```diff
@@ -0,0 +1,1378 @@
+            "Attention",
+            inputs=["input", "weight", "bias", "mask_index"],
+            outputs=["output", "present"],
+            domain="com.microsoft",
+            num_heads=num_heads,
+        )
+
+        present_output_shape = (2, batch_size, num_heads, sequence_length, head_size)
+
+        input_shape = list(input_.shape)
+        weight_shape = list(weight.shape)
+        bias_shape = list(bias.shape)
+        mask_shape = list(mask_index.shape)
+        output_shape = list(input_.shape)
+        present_shape = list(present_output_shape)
+        if dynamic:
+            input_shape = ["?" for _ in range(len(input_.shape))]
+            weight_shape = ["?" for _ in range(len(weight.shape))]
+            bias_shape = ["?" for _ in range(len(bias.shape))]
+            mask_shape = ["?" for _ in range(len(mask_index.shape))]
+            output_shape = ["?" for _ in range(len(input_.shape))]
+            present_shape = ["?" for _ in range(len(present_output_shape))]
+
+        graph = helper.make_graph(
+            [node],
+            "attention_test",
+            inputs=[
+                helper.make_tensor_value_info("input", TensorProto.FLOAT, input_shape),
+                helper.make_tensor_value_info("weight", TensorProto.FLOAT, weight_shape),
+                helper.make_tensor_value_info("bias", TensorProto.FLOAT, bias_shape),
+                helper.make_tensor_value_info("mask_index", TensorProto.INT32, mask_shape),
+            ],
+            outputs=[
+                helper.make_tensor_value_info("output", TensorProto.FLOAT, output_shape),
+                helper.make_tensor_value_info("present", TensorProto.FLOAT, present_shape),
+            ],
+        )
+
+        model = helper.make_model(graph, producer_name="attention_test")
+
+        check_correctness(
+            model,
+            inputs={"input": input_, "weight": weight, "bias": bias, "mask_index": mask_index},
+        )
+        # "present" output should be nullptr when the "past" input isn't included,
+        # but ort requires an output shape to be specified?
+        # verify_with_ort_with_inputs(
+        #     model,
+        #     [input_, weight, bias, mask_index],
+        #     [input_.shape, present_output_shape],
+        #     target=target,
+        #     dev=dev,
+        #     rtol=1e-4,
+        #     atol=1e-4,
+        # )
+
+    hidden_size = 384
+    batch_size = 4
+    sequence_length = 4
+    num_heads = 12
+    head_size = 32
+
+    dtype = "float32"
+    input_array = np.random.random((batch_size, sequence_length, hidden_size)).astype(dtype)
+    weight = np.random.normal(size=(hidden_size, 3 * hidden_size)).astype(dtype) * 0.1
+    bias = np.random.randn(3 * hidden_size).astype(dtype)
+    mask_index = np.full((batch_size, sequence_length), 1).astype("int32")
+
+    verify_attention(input_array, weight, bias, mask_index, num_heads)
+
+
+@pytest.mark.parametrize("dynamic", [True, False])
+def test_pad(dynamic):
```

Comment:
What is the difference between this test and the one below it?

Reply:
I removed this test